app.py

```diff
@@ -86,12 +106,95 @@ def delete(id):
         return redirect(url_for('history'))
 
 
+#Database API for getting charts
+@app.route('/request/chart/<id>', methods=['POST', 'GET'])
+def updateChart(id):
+    getDate = records.query.order_by()
+    toAppend = []
+    for n in getDate:
+        toAppend.append(n.time)
+    sortedArray = sorted(toAppend, key=sortingDates)
+    #function to add up all records by id
+    sumArray = []
+    numCache = []
+    finalCache = []
+    for t in range(0, len(sortedArray)):
+        if t == 0:
+            tofilter = records.query.filter_by(time = sortedArray[t])
+            for n in tofilter:
+                finalCache.append(n.operator)
+                finalCache.append(n.record)
+                numCache.append(n.operator)
+                numCache.append(n.record)
+            sumArray.append(addUp(numCache))
+            numCache.clear()
+            toCheck = sortedArray[t]
+        if toCheck == sortedArray[t]:
+            continue
+        else:
+            tofilter = records.query.filter_by(time = sortedArray[t])
+            for n in tofilter:
+                finalCache.append(n.operator)
+                finalCache.append(n.record)
+                numCache.append(n.operator)
+                numCache.append(n.record)
+            sumArray.append(addUp(numCache))
+            numCache.clear()
+            toCheck = sortedArray[t]
+    final = addUp(finalCache)
+    databaseDataQuery = sortedArray
+    databaseTrimVersion = list(dict.fromkeys(databaseDataQuery))
+    allRecords = records.query.order_by()
+    if id == "dealy":
+        return jsonify('', render_template('db.html', sortedArray = sortedArray, sumArray = sumArray, databaseTrimVersion = 
+    databaseTrimVersion, allRecords = allRecords, final = final))
+    elif id == "monthly":
+        #loop for monthly
+        stringArray = databaseTrimVersion
+        intArray = sumArray
+        stringToAppend = []
+        intToAppend = []
+        sum = 0
+        for n in range(0, len(stringArray)):
+            if n == 0:
+                toCheck = stringArray[n][5:7]
+                sum = sum + float(intArray[n])
+                stringToAppend.append(stringArray[n][0:7])
+            elif stringArray[n][5:7] == toCheck:
+                sum = sum + float(intArray[n])
+            elif stringArray[n][5:7] != toCheck:
+                stringToAppend.append(stringArray[n][0:7])
+                intToAppend.append(sum)
+                sum = 0
+                toCheck = stringArray[n][5:7]
+                sum = sum + float(intArray[n])
+        intToAppend.append(sum)
+        sumArray = intToAppend
+        databaseTrimVersion = stringToAppend
+        return jsonify('', render_template('db.html', sortedArray = sortedArray, sumArray = sumArray, databaseTrimVersion = databaseTrimVersion, allRecords = allRecords, final = final))
```

Comment:
Das ist nicht sehr cool, das erst als HTML zu rendern und dann in ein JSON Objekt zu packen. Da solltest du nur deine Daten in JSON packen und das Aussehen etc. im Frontend machen.
Aktuell erfüllt das allerdings seinen Zweck, da du dir dann Javascript Code sparst. Ist nur nicht sehr schön. Bei einem größeren Projekt, würdest du das allerdings nicht machen. Der Vorteil von den Daten als JSON ist, dass du damit eine Schnittstelle für andere Tools bereit stellen könntest.